scr/main/java/bittercode/util/StoreUtil.java

```diff
@@ -0,0 +1,86 @@
+    }
+
+    /**
+     * Modify the active tab in the page menu bar
+     */
+    public static void setActiveTab(PrintWriter pw, String activeTab) {
+
+        pw.println("<script>document.getElementById(activeTab).classList.remove(\"active\");activeTab=" + activeTab
+                + "</script>");
+        pw.println("<script>document.getElementById('" + activeTab + "').classList.add(\"active\");</script>");
+
+    }
+
+    /**
+     * Add/Remove/Update Item in the cart using the session
+     */
+    public static void updateCartItems(HttpServletRequest req) {
+        String selectedBookId = req.getParameter("selectedBookId");
+        HttpSession session = req.getSession();
+        if (selectedBookId != null) { // add item to the cart
+
+            // Items will contain comma separated bookIds that needs to be added in the cart
+            String items = (String) session.getAttribute("items");
+            if (req.getParameter("addToCart") != null) { // add to cart
+                if (items == null || items.length() == 0)
+                    items = selectedBookId;
+                else if (!items.contains(selectedBookId))
+                    items = items + "," + selectedBookId; // if items already contains bookId, don't add it
+
+                // set the items in the session to be used later
+                session.setAttribute("items", items);
+
+                /*
+                 * Quantity of each item in the cart will be stored in the session as:
+                 * Prefixed with qty_ following its bookId
+                 * For example 2 no. of book with id 'myBook' in the cart will be
+                 * added to the session as qty_myBook=2
+                 */
+                int itemQty = 0;
+                if (session.getAttribute("qty_" + selectedBookId) != null)
+                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
+                itemQty += 1;
+                session.setAttribute("qty_" + selectedBookId, itemQty);
+            } else { // remove from the cart
+                int itemQty = 0;
+if (session.getAttribute("qty_" + selectedBookId) != null)
+                    itemQty = (int) session.getAttribute("qty_" + selectedBookId);
+                if (itemQty > 1) {
+                    itemQty--;
+                    session.setAttribute("qty_" + selectedBookId, itemQty);
+                } else {
+                    session.removeAttribute("qty_" + selectedBookId);
+                    items = items.replace(selectedBookId + ",", "");
+                    items = items.replace("," + selectedBookId, "");
+                    items = items.replace(selectedBookId, "");
+                    session.setAttribute("items", items);
+                }
+                else  {
+                    session.removeAttribute("qty_" + selectedBookId);
+                    session.setAttribute("items", items.replace(selectedBookId, ""));
+                }
+            }
+        }
+
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider splitting updateCartItems into smaller methods</b></div>
<div id="fix">

The `updateCartItems` method appears to be quite lengthy and handles multiple responsibilities (adding/removing items, managing quantities). Consider breaking it down into smaller focused methods like `addToCart` and `removeFromCart` to improve readability and maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -37,49 +37,65 @@
 public static void updateCartItems(HttpServletRequest req) {
     String selectedBookId = req.getParameter("selectedBookId");
     HttpSession session = req.getSession();
     if (selectedBookId != null) {
         if (req.getParameter("addToCart") != null) {
             addToCart(selectedBookId, session);
         } else {
             removeFromCart(selectedBookId, session);
         }
     }
 }

 private static void addToCart(String selectedBookId, HttpSession session) {
     String items = (String) session.getAttribute("items");
     if (items == null || items.length() == 0)
         items = selectedBookId;
     else if (!items.contains(selectedBookId))
         items = items + "," + selectedBookId;
     session.setAttribute("items", items);
     updateItemQuantity(selectedBookId, session, true);
 }

 private static void removeFromCart(String selectedBookId, HttpSession session) {
     updateItemQuantity(selectedBookId, session, false);
 }

 private static void updateItemQuantity(String selectedBookId, HttpSession session, boolean isAdd) {
     int itemQty = 0;
     if (session.getAttribute("qty_" + selectedBookId) != null)
         itemQty = (int) session.getAttribute("qty_" + selectedBookId);
     if (isAdd) {
         itemQty += 1;
         session.setAttribute("qty_" + selectedBookId, itemQty);
     } else {
         handleItemRemoval(selectedBookId, session, itemQty);
     }
 }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2517477681>#c23dbc</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged